src/NuGet.Services.AzureSearch/Owners2AzureSearch/IOwnerDataClient.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using NuGetGallery;
+
+namespace NuGet.Services.AzureSearch.Owners2AzureSearch
+{
+    public interface IOwnerDataClient
+    {
+        Task<ResultAndAccessCondition<SortedDictionary<string, SortedSet<string>>>> ReadLatestIndexedAsync();
+        Task ReplaceLatestIndexedAsync(
+            SortedDictionary<string, SortedSet<string>> newData,
+            IAccessCondition accessCondition);
+        Task UploadChangeHistoryAsync(IEnumerable<string> packageIds);
```

Comment:
Can the `IEnumerable` be an `IReadOnlyList`? I've been bitten too many times by LINQ :)

Reply:
Same comment with line 25 of `PackageIdToOwnersBuilder.cs`

Reply:
`IEnumerable` is not LINQ per se and should be acceptable as a method parameter. It's the implementor's responsibility to enumerate it an appropriate number of times. In 99% of our code there is a single implementation of these interfaces so it's easy to get right in the one place it needs to. But sure, I don't care much so I'll switch it.